File: src/main/java/com/github/joelws/release/tracker/service/release/execution/ListReleaseServiceExecution.java
package com.github.joelws.release.tracker.service.release.execution;

import com.github.joelws.release.tracker.dao.release.ReleaseDao;
import com.github.joelws.release.tracker.entity.release.Release;
import com.github.joelws.release.tracker.service.ServiceExecution;

import java.util.List;

public class ListReleaseServiceExecution implements ServiceExecution<Void, List<Release>> {

    private final ReleaseDao releaseDao;

    public ListReleaseServiceExecution(ReleaseDao releaseDao) {
        this.releaseDao = releaseDao;
    }

    @Override
    public List<Release> execute(final Void param) {
        return releaseDao.list();
    }
}

File: src/main/kotlin/com/github/joelws/release/tracker/service/release/execution/ListReleaseServiceExecution.kt
package com.github.joelws.release.tracker.service.release.execution

import com.github.joelws.release.tracker.dao.release.ReleaseDao
import com.github.joelws.release.tracker.entity.release.Release
import com.github.joelws.release.tracker.service.ServiceExecution

open class ListReleaseServiceExecution(private val releaseDao: ReleaseDao) : ServiceExecution<Nothing?, List<Release>> {

    override fun execute(param: Nothing?): List<Release> = releaseDao.list()
}
